### PR TITLE
refactor NewCmdJoin function

### PR DESCRIPTION
**Pull request by wackxu**
**What this PR does / why we need it**:
refactor NewCmdJoin function to make the code more readable and concision
1、split the function into several small function 
2、extract the big variable from the function

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by luxas** [comment]
/assign @fabriziopandini 
PTAL and give your LGTM when ready
/ok-to-test

**Comment by luxas** [comment]
Thank you @wackxu for this contribution!

File: cmd/kubeadm/app/cmd/join.go [hunk 1]
```diff
@@ -49,20 +50,8 @@ var (
...
-	api.Scheme.Default(cfg)
-
-	var skipPreFlight bool
-	var cfgPath string
 
-	cmd := &cobra.Command{
-		Use:   "join <flags> [DiscoveryTokenAPIServers]",
-		Short: "Run this on any machine you wish to join an existing cluster",
-		Long: dedent.Dedent(`
+	joinLong = dedent.Dedent(`
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/cmd/join.go]: nit joinLongDescription

File: cmd/kubeadm/app/cmd/join.go [hunk 2]
```diff
@@ -102,7 +91,21 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
...
+// NewCmdJoin returns "kubeadm join" command.
+func NewCmdJoin(out io.Writer) *cobra.Command {
+	cfg := &kubeadmapiext.NodeConfiguration{}
+	api.Scheme.Default(cfg)
+
+	var skipPreFlight bool
+	var cfgPath string
+
+	cmd := &cobra.Command{
+		Use:   "join <flags> [DiscoveryTokenAPIServers]",
```
Comment (fabriziopandini) [inline cmd/kubeadm/app/cmd/join.go]: This is not part of this PR, but might be you can fix...

The output of this line is now:
```kubeadm join <flags> [DiscoveryTokenAPIServers] [flags]```

Could you kindly remove `<flags>`?

**Comment by fabriziopandini** [comment]
The approach implement by this PR is consistent with what already used for e.g. init. and basically looks good to me; I added two nit, but not blocking.

It would be nice to get the same approach applied consistently to all the commands, might be sharing `Add..Flags` functions across commands when feasible. But this is clearly out of scope of this PR.

/LGMT

File: cmd/kubeadm/app/cmd/join.go [hunk 3]
```diff
@@ -102,7 +91,21 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
...
+// NewCmdJoin returns "kubeadm join" command.
+func NewCmdJoin(out io.Writer) *cobra.Command {
+	cfg := &kubeadmapiext.NodeConfiguration{}
+	api.Scheme.Default(cfg)
+
+	var skipPreFlight bool
+	var cfgPath string
+
+	cmd := &cobra.Command{
+		Use:   "join [DiscoveryTokenAPIServers] [flags]",
```
Comment (luxas) [inline cmd/kubeadm/app/cmd/join.go]: remove the `[DiscoveryTokenAPIServers]` here please (I know it was there before though)

**Review by luxas — COMMENTED** [review]
super small nit still.
Otherwise LGTM

File: cmd/kubeadm/app/cmd/join.go (hunk 3 above)
Comment (wackxu) [reply cmd/kubeadm/app/cmd/join.go]: @luxas Thanks for your review, I have fix it.

**Review by luxas — APPROVED** [review]
/lgtm
/approve no-issue
